test/run_translated_c.zig

```diff
@@ -268,5 +268,73 @@ pub fn addCases(cases: *tests.RunTranslatedCContext) void {
         \\    if (count != 4) abort();
         \\    return 0;
         \\}
-    ,"");
+    , "");
+
+    cases.add("array value type casts properly",
+        \\#include <stdlib.h>
+        \\const unsigned int choose[53][10];
+        \\static int hash_binary(int k)
+        \\{
+        \\    int sum = 0;
+        \\    sum += choose[0][k];
+        \\    return sum;
+        \\}
+        \\
+        \\int main() {
+        \\    hash_binary(4);
```

Comment:
Could you make these tests actually assert that the correct value was reached?

Reply:
Done, had to remove the 'const' though from the array so the example is slightly different than the original ticket but that should be fine. Didn't check it initially since it was uninitialized and the value could've been anything